src/ConfigMap.cpp

```diff
@@ -79,6 +65,36 @@ namespace PCOE {
         loadArguments(argc, argv);
     }
 
+    const std::vector<std::string>& ConfigMap::getVector(const std::string key) const {
+        return map.at(key);
+    }
+
+    std::vector<std::string> &ConfigMap::getVector(const std::string key) {
+        return map[key];
+    }
+
+    double ConfigMap::getDouble(const std::string key) const {
+        if (map.at(key).size() > 1) {
+            throw std::invalid_argument("Size of vector at " + key + " is not 1.");
```

Comment:
I'm not sure if invalid_argument is the right error here. It might be, but lets check that there isn't a better fitting exception class

Reply:
Also can we make the error message "Cannot get " + key + " as scalar value, is a vector" or something along those lines. The message you have here doesn't make it clear why a vector with size > 1 is a problem.